### PR TITLE
Refactoring testing

### PR DESCRIPTION
**Pull request by mattgrill**
- `npm i` to update dependencies. 
- `npm t` to check tests.
- `npm run coverage` to check coverage.

Moves away from a tightly coupled relationship between entities and entity handlers. A single generic entity handler accepts parameters and configures what it needs. Tests are also much faster, 💯

### CODE REVIEW

File: lib/entity.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+const Request = require('./helpers/request');
+const methods = require('./helpers/methods');
+
+module.exports = class Entity extends Request {
```
Comment (wimleers) [inline lib/entity.js:4]: `Entity` sounds like the correct name to me :)

File: lib/hydrant.js [hunk 2]
```diff
@@ -14,10 +11,10 @@ module.exports = class Hydrant {
...
   constructor(base, credentials) {
     this.api = {
-      node: new Node(base, credentials),
-      menu: new Menu(base, credentials),
-      contentType: new ContentType(base, credentials),
-      comment: new Comment(base, credentials),
+      content: new Entity(base, credentials, {get: 'node', set: 'node', create: 'entity/node', delete: 'node'}),
+      contentType: new Entity(base, credentials, {get: '/entity/node_type', set: '/entity/node_type', create: '/entity/node_type', delete: '/entity/node_type'}),
+      menu: new Entity(base, credentials, {get: 'entity/menu', set: 'entity/menu', create: 'entity/menu', delete: 'entity/menu'}),
+      comment: new Entity(base, credentials, {get: 'comment', set: 'comment', create: 'comment', delete: 'comment'}),
```
Comment (wimleers) [inline lib/hydrant.js]: lgtm

File: lib/entity.js (hunk 1 above)
Comment (mattgrill) [inline lib/entity.js:4]: @wimleers Thanks. At some point I will probably roll the `entity query` work into this, so we don't have two separate ways to get items.

File: lib/hydrant.js [hunk 3]
```diff
@@ -13,13 +10,54 @@ module.exports = class Hydrant {
...
-      comment: new Comment(base, credentials),
+      content: new Entity(base, credentials, {get: 'node', set: 'node', create: 'entity/node', delete: 'node'}),
+      comment: new Entity(base, credentials, {get: 'comment', set: 'comment', create: 'comment', delete: 'comment'}),
+      contentType: new Entity(base, credentials, {get: '/entity/node_type', set: '/entity/node_type', create: '/entity/node_type', delete: '/entity/node_type'}),
+      file: new Entity(base, credentials, {get: 'entity/file', set: 'entity/file', create: 'entity/file', delete: 'entity/file'}),
+      menu: new Entity(base, credentials, {get: 'entity/menu', set: 'entity/menu', create: 'entity/menu', delete: 'entity/menu'}),
+      taxonomyTerm: new Entity(base, credentials, {get: 'taxonomy/term', set: 'taxonomy/term', create: 'taxonomy/term', delete: 'taxonomy/term'}),
+      taxonomyVocabulary: new Entity(base, credentials, {get: 'entity/taxonomy_vocabulary', set: 'entity/taxonomy_vocabulary', create: 'entity/taxonomy_vocabulary', delete: 'entity/taxonomy_vocabulary'}),
+      user: new Entity(base, credentials, {get: 'user', set: 'user', create: 'user', delete: 'user'}),
```
Comment (prestonso) [inline lib/hydrant.js:23]: w00t

File: test/entityQuery.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  const entityQuery = new EntityQuery('http://foo.dev', {user: 'a', pass: 'b'}, 'node');
+
+  entityQuery.range(0, 5);
+
+  t.is('range[start]=0&range[length]=5', entityQuery.entityQuery.getQueryString(), 'Unexpected response.');
+  t.end();
+
+});
+
+test.cb('Contion', t => {
```
Comment (prestonso) [inline test/entityQuery.js]: `'Condition'`

**Comment by prestonso** [comment]
Nice work! Just had one small nit.

**Comment by prestonso** [comment]
elijah-wood-i-love-this

lgtm :shipit:

**Pull request merged**